### PR TITLE
docs(TaskList): Include taskFilterByVariable in the API Docs - (alpha5 release)

### PR DESCRIPTION
**Pull request by nathansandi**
## Description

Include the filter option `taskVariables`, relater to Filter Task by Variables over our API 

Parent ticket: 
https://github.com/orgs/camunda/projects/41/views/2?pane=issue&itemId=28998533

This is gonne be merged after alpha 5 release 

## When should this change go live?

- [ ] This change is not yet live and should not be merged until 08/09/2023 (apply `hold` label or convert to draft PR)?

## PR Checklist

- [x] I have added changes to the relevant `/versioned_docs` directory, or they are not for an **already released version**.
- [x] I have added changes to the main `/docs` directory (aka `/next/`), or they are not for **future versions**.
- [x] My changes require an [Engineering review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned an engineering manager or tech lead as a reviewer, or my changes do not require an Engineering review.
- [ ] My changes require a [technical writer review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @christinaausley as a reviewer, or my changes do not require a technical writer review.

### CODE REVIEW

**Review by christian-konrad — CHANGES_REQUESTED** [review]
There are 2 small changes we should apply, thanks!

File: docs/apis-tools/tasklist-api-rest/schemas/requests/task-search-request.mdx [hunk 1]
```diff
@@ -156,6 +164,10 @@ TaskSearchRequest - query object to search tasks by provided params.
 
 > Specifying a range of due dates for the tasks to search for.
 
+#### [<code style={{ fontWeight: 'normal' }}>TaskSearchRequest.<b>taskVariables</b></code>](#)<Bullet />[`TaskVariables`](../models/task-variables.mdx) <Badge class="secondary" text="list"/>
+
+> An array of objects specifying the variables that should be filtered plus the operator. This is limited to truncated variables only.
```
[the remark this answers is not in the document]
Comment (nathansandi) [reply docs/apis-tools/tasklist-api-rest/schemas/requests/task-search-request.mdx:169]: We can explain here, 

WDYT About this:
`The query allows for an array of filter clauses to specify which variables to filter for. If these clauses are defined, the query will return only the tasks that meet all the specified conditions. However, it's important to note that this filtering mechanism is designed to work exclusively with truncated variables. This means that variables of a larger size are not compatible with this filter, and attempts to use them may result in inaccurate or incomplete query results.`

File: docs/apis-tools/tasklist-api-rest/schemas/requests/task-search-request.mdx [hunk 2]
```diff
@@ -156,6 +164,10 @@ TaskSearchRequest - query object to search tasks by provided params.
 
 > Specifying a range of due dates for the tasks to search for.
 
+#### [<code style={{ fontWeight: 'normal' }}>TaskSearchRequest.<b>taskVariables</b></code>](#)<Bullet />[`TaskVariables`](../models/task-variables.mdx) <Badge class="secondary" text="list"/>
+
+> An array of filter clauses specifying the variables to filter for.  If defined, the query returns only tasks to which all clauses apply.. However, it's important to note that this filtering mechanism is designed to work exclusively with truncated variables. This means that variables of a larger size are not compatible with this filter, and attempts to use them may result in inaccurate or incomplete query results.
```
Comment (christian-konrad) [inline docs/apis-tools/tasklist-api-rest/schemas/requests/task-search-request.mdx:169]: Typo (two ".." after apply), otherwise good!

**Review by christian-konrad — DISMISSED** [review]
Small typo, please fix before merge, otherwise good!

File: docs/apis-tools/tasklist-api-rest/schemas/requests/task-search-request.mdx (hunk 2 above)
Comment (nathansandi) [reply docs/apis-tools/tasklist-api-rest/schemas/requests/task-search-request.mdx]: Just fixed :) Can you reapprove ? Thanks

**Review by christian-konrad — DISMISSED** [review]

**Comment by akeller** [comment]
This needs to be merged and release for tomorrow's alpha, correct?

**Comment by nathansandi** [comment]
Yes :)

**Pull request merged**